test/components/util/util.go

```diff
@@ -321,3 +322,27 @@ func (p *PortForwardInfo) WaitUntilForwardingAvailable(t *testing.T) {
 	}
 	p.findLocalPort(t)
 }
+
+// IsPlatformSupported takes in the test object and the list of supported platforms. The function
+// detects the supported platform from environment variable. And if the platform is available in the
+// supported platforms provided then this returns true otherwise false. If the supported platforms
+// list is empty it is interpreted as all platforms are supported.
+func IsPlatformSupported(t *testing.T, platforms string) bool {
+	// This means that all platforms are supported.
+	if platforms == "" {
+		return true
+	}
+
+	// Find out what platform we are running on.
+	platform := os.Getenv("PLATFORM")
+	if platform == "" {
+		t.Fatal("env var PLATFORM was not set")
+	}
+
+	// The platform should be present in the list of supported platforms.
+	if strings.Contains(platforms, platform) {
```

Comment:
What about `packet` and `packer-arm`? If run on `packet` and test is only intended to run on `packet-arm`, it will be incorrectly triggered.

Reply:
Also it could be just `return strings.Contains(platforms, platform)`.

Reply:
Yeah this is possible, but all the tests that run on packet-arm run on packet. It is super set of the other.

Reply:
This may produce unexpected behavior in some cases, that's not good :/